edb/testbase/server.py

```diff
@@ -1806,3 +1814,27 @@ def get_cases_by_shard(cases, selected_shard, total_shards, verbosity, stats):
               f' / {int(total_est / 60)}m {int(total_est % 60)}s, '
               f'{len(setups)}/{setup_count} databases to setup.')
     return cases
+
+
+def find_available_port(port_range=(49152, 65535), max_tries=1000):
```

Comment:
I think we already have this function defined somewhere. Also, perhaps, instead of a loop, it should just create a tmp socket pair (or create a socket and bind to 0) and reuse its port.

Reply:
> I think we already have this function defined somewhere.

Right, it was removed in [#2251](https://github.com/edgedb/edgedb/pull/2251/files#diff-04df2656d0211c7f0c32b3ae2705297d32ea26bcdb7c4f9f1282b3a9016eb25dL43).

> create a tmp socket pair (or create a socket and bind to 0) and reuse its port.

I'm not sure what you meant by socket pair (they are UNIX sockets right?), but I'm changing the method into "bind to 0".

Reply:
> I'm not sure what you meant by socket pair (they are UNIX sockets right?), 

You can pass AF_INET family to `socketpair()`.

> but I'm changing the method into "bind to 0".

Whatever is simpler.